test/cabal-22/default.nix

```diff
@@ -25,7 +25,7 @@ in
 
       printf "checking that executable is dynamically linked to system libraries... " >& 2
     '' + optionalString stdenv.isLinux ''
-      ldd $exe | grep libpthread
+      ldd $exe | grep libgmp
```

Comment:
Hmm, we may link libgmp statically as well (if we agree to release it as GPL software I believe).

I don't think we'll need to link against anything dynamically on linux, if we so wanted to.

Reply:
Fully static linking will be a future PR... Personally I like the GPL but all IOHK projects need to be Apache License 2.0. So that rules out gmp. I believe we need to add an option to Haskell.nix so that `haskell.compiler.integer-simple.ghc865` can be used.